pulser-core/pulser/channels/base_channel.py

```diff
@@ -363,11 +362,11 @@ def validate_pulse(self, pulse: Pulse) -> None:
                 "The pulse's detuning values go out of the range "
                 "allowed for the chosen channel."
             )
-        amp_area = pulse.amplitude.integral
-        if 0 < amp_area < self.min_amp_area:
+        avg_amp = np.average(pulse.amplitude.samples)
+        if 0 < avg_amp < self.min_avg_amp:
```

Comment:
So `avg_amp` can be zero ? That's to realize pulse of 0 amplitude I guess. I thought this avg_amp was to be understood as an offset, so only when light goes through (if pulse is of zero amplitude, there is no offset ?)

Reply:
Well, we still need to do things like detuned delays, so we have to allow pulses at 0 amplitude. There might be some noise there, but it is the same as there would be in a delay.

Reply:
Yes I was thinking about detuned delays and delay when speaking about "pulse of 0 amplitude". One thing we could do is convert them into pulse having a constant amplitude equal to min_avg_amp, but I guess that would be going a bit too far...

Reply:
Definitely too far